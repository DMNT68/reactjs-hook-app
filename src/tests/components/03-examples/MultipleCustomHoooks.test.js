import React from 'react';
import { shallow } from 'enzyme';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
import { MultipleCustomHokks } from '../../../components/03-examples/MultipleCustomHokks';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');


describe('Pruebas en <MultipleCustomHokks />', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });

    
    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHokks /> );
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should debe de mostrar la informaciÃ³n', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Fernando',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHokks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-3').text().trim() ).toBe( 'Hola Mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Fernando' );

    })
    
    

})