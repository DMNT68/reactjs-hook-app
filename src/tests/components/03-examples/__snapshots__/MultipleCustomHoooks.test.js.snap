// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHokks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    />
    <div>
      Loading...
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
